code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -3140,6 +3140,36 @@
 	icon = 'icons/obj/drinks/mixed_drinks.dmi'
 	icon_state = "pina_colada"
 
+/datum/reagent/consumable/ethanol/pina_olivada
+	name = "Piña Olivada"
+	description = "An oddly designed concoction of olive oil and pineapple juice."
+	boozepwr = 20 // the oil coats your gastrointestinal tract, meaning you can't absorb as much alcohol. horrifying
+	color = "#493c00"
+	quality = DRINK_NICE
+	taste_description = "a horrible embolism of pineapple and olive oil"
+
+/datum/reagent/consumable/ethanol/pina_olivada/on_mob_life(mob/living/carbon/drinker, delta_time, times_fired)
+	if(prob(90))
+		return ..() // first must clear 10% chance that something funny happens
```

Comment:
except the actual chances for something funny to happen are 0.15%/tick for the first effect and 0.05%/tick for the second effect, because you put prob()s on those as well

you also aren't accounting for delta_time

Reply:
that's fine